packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_update_loan_token.test.ts

```diff
@@ -0,0 +1,286 @@
+      tokenTx: loanTokenId
+    }, script)
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow('DeFiDRpcError: \'LoanUpdateLoanTokenTx: token symbol should be non-empty and starts with a letter (code 16)\', code: -26')
+  })
+
+  it('should not updateLoanToken if the symbol is used in other token', async () => {
+    await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token4',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.container.call('setloantoken', [{
+      symbol: 'Token4',
+      name: 'Token4',
+      priceFeedId: 'Token4/USD',
+      mintable: true,
+      interest: new BigNumber(0.06)
+    }, []])
+    await testing.generate(1)
+
+    await fundEllipticPair(testing.container, providers.ellipticPair, 10) // Fund 10 DFI UTXO
+    await providers.setupMocks() // Required to move utxos
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.updateLoanToken({
+      symbol: 'Token1',
+      name: 'Token4',
+      currencyPair: { token: 'Token2', currency: 'USD' },
+      mintable: true,
+      interest: new BigNumber(0.07),
+      tokenTx: loanTokenId
+    }, script)
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow('DeFiDRpcError: \'LoanUpdateLoanTokenTx: token with key \'Token1\' already exists! (code 16)\', code: -26')
+  })
+
+  it('should updateLoanToken if name is more than 128 letters', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.updateLoanToken({
+      symbol: 'Token2',
+      name: 'x'.repeat(129), // 129 letters,
+      currencyPair: { token: 'Token2', currency: 'USD' },
+      mintable: true,
+      interest: new BigNumber(0.08),
+      tokenTx: loanTokenId
+    }, script)
+    await sendTransaction(testing.container, txn)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('x'.repeat(128)) // Only remain the first 128 letters
+  })
+
+  it('should updateLoanToken if two loan tokens have the same name', async () => {
+    await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token5',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'Token5',
+      name: 'Token5',
+      priceFeedId: 'Token5/USD'
+    })
+    await testing.generate(1)
+
+    await fundEllipticPair(testing.container, providers.ellipticPair, 10) // Fund 10 DFI UTXO
+    await providers.setupMocks() // Required to move utxos
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.updateLoanToken({
+      symbol: 'Token1',
+      name: 'Token1',
```

Comment:
```suggestion
      name: 'Token5',
```
?

Reply:
Fixed

Reply:
Actually `Token1` is correct. I change another token to have same name.